test/selectors/dashboard-selectors_spec.js

```diff
@@ -60,23 +64,44 @@ describe('dashboard selectors', () => {
     })
   })
 
-  describe('getSortedStudents', () => {
-    describe('when sortBy = "name"', () => {
-      const state = fromJS({
-        report: {
-          students: { 1: s1, 2: s2, 3: s3 }
-        },
-        dashboard: {
-          sortBy: 'name'
+  describe('getStudentTotalProgress', () => {
+    it('should return hash with student total progress', () => {
+      expect(getStudentTotalProgress(state({})).toJS()).to.eql(
+        {
+          1: 1.5,
+          2: 1.5,
+          3: 0
         }
-      })
+      )
+    })
+  })
 
+  describe('getSortedStudents', () => {
+    describe('when sorting by name', () => {
       it('should return sorted list of students', () => {
-        expect(getSortedStudents(state).toJS()).to.eql(
+        expect(getSortedStudents(state({})).toJS()).to.eql(
```

Comment:
Because the sorting selector now pulls in student progress, I needed to use the bigger state from above, which makes for a much messier test. Do you know if there's a way to stub out selectors so I can get away with passing a smaller state?

Reply:
That's a good question. I don't know yet. I don't think it's too bad here though, as we shared this state between all the sorting methods. But it might get useful in the future.